os/arch/arm/src/s5j/s5j_i2s.c

```diff
@@ -1600,18 +1600,20 @@ static int i2s_pause(struct i2s_dev_s *dev)
 {
 	struct s5j_i2s_s *priv = (struct s5j_i2s_s *)dev;
 	irqstate_t flags;
+	volatile u32 reg = 0;
 	DEBUGASSERT(priv);
 
+	reg = getreg32(priv->base + S5J_I2S_CON);
 #ifdef I2S_HAVE_TX_P
-	if (priv->txpenab) {
+	if (priv->txpenab && reg & I2S_CR_TXDMACTIVE) {
```

Comment:
Why this check of DMA active status? What happens if pause bit is set when dma is inactive? Any harm? Better you check the status within irqsave.

Reply:
We need to pause DMA channel, only if it is currently active.
Not doing this was resulting in UTC getting blocked in pcm_wait() which waits for a dequeue msg.

I modified to use irqsave and perform the entire pause / resume as one single atomic operation.